samples: cleanup sysbuild/hello_world sample

tejlmand opened the pull request:
Remove the prompt for REMOTE_BOARD.
REMOTE_BOARD should not be having a prompt or be configurable on command line.
Instead REMOTE_BOARD should be defined based on the SoC as this will allow creation of new boards using the same SoC, and thereby be able to build the sample out-of-the-box for any supported SoC.

This allows us to remove several single line config files.

For SoCs with more than two CPU clusters, such as the nRF54h20, then a choice is provided to select specific core.

Remove the REMOTE_BOARD restriction, as this sample will build and run even for single core SoCs, and may be useful for testing other sysbuild multi-image features even for single cores SoCs.

tejlmand committed:
samples: cleanup sysbuild/hello_world sample

Remove the prompt for REMOTE_BOARD.
REMOTE_BOARD should not be having a prompt or be configurable on command
line.
Instead REMOTE_BOARD should be defined based on the SoC as this will
allow creation of new boards using the same SoC, and thereby be able
to build the sample out-of-the-box for any supported SoC.

This allows us to remove several single line config files.

For SoCs with more than two CPU clusters, such as the nRF54h20, then a
choice is provided to select specific core.

Remove the REMOTE_BOARD restriction, as this sample will build and run
even for single core SoCs, and may be useful for testing other sysbuild
multi-image features even for single cores SoCs.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

sylvioalves commented:
I wonder whether we can have `config REMOTE_BOARD` at `zephyr/boards/Kconfig` and have it properly set in each soc (or board) configuration by default. That might clean up this `Kconfig.sysbuild` list and eventually in other samples like openamp, rpmsg and so.

tejlmand committed:
samples: pass on board revision in sysbuild/hello_world

Pass on the board revision to remote image build if specified for the
application image board target.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand commented on samples/sysbuild/hello_world/sysbuild.cmake:
```diff
@@ -1,15 +1,13 @@
+    APPLICATION remote
+    SOURCE_DIR ${APP_DIR}/remote
+    BOARD ${SB_CONFIG_REMOTE_BOARD}
+  )
```
considering this a cleanup, then it's technically not missing :wink: 

But added extra commit which fixes this in addition.

sylvioalves commented:
> Today, only SoCs explicitly specified in the `REMOTE_BOARD` are supported, but having a default for 90% of our SoCs would give better experience out-of-the-box.

.. and eventually getting rid of  .conf and .overlay files.

Thank you all for clearing this up. I know it is out of scope for this PR.

pdgendt reviewed (dismissed):
Needs a rebase

nordic-segl commented on samples/sysbuild/hello_world/sysbuild.cmake:
```diff
@@ -1,15 +1,14 @@
-  SOURCE_DIR ${APP_DIR}/remote
-  BOARD ${SB_CONFIG_REMOTE_BOARD}
-)
+if(DEFINED SB_CONFIG_REMOTE_BOARD)
```
Is `DEFINED` needed here? I'm asking because I sow other PR that (besides other things) was removing this.
